pulp_smash/pulp_smash_cli.py

```diff
@@ -0,0 +1,177 @@
+    system_hostname = click.prompt('System hostname')
+    if click.confirm('Is Pulp published using HTTP?'):
+        system_api_scheme = 'http'
+    else:
+        system_api_scheme = 'https'
+
+    if system_api_scheme == 'https' and click.confirm('Verify HTTPS?'):
+        certificate_path = click.prompt('SSL certificate path', default='')
+        if not certificate_path:
+            system_api_verify = True  # pragma: no cover
+        else:
+            system_api_verify = certificate_path
+    else:
+        system_api_verify = False
+
+    if click.confirm('Is Pulp\'s message broker Qpid?', default=True):
+        amqp_broker = 'qpidd'
+    else:
+        amqp_broker = 'rabbitmq'
+    using_ssh = not click.confirm(
+        'Are you running Pulp Smash on the Pulp system?')
+    if using_ssh:
+        click.echo(
+            'Pulp Smash will be configured to access the Pulp system using '
+            'SSH, because that, some additional information will be required.'
+        )
+        ssh_user = click.prompt('SSH username to use', default='root')
+        click.echo(
+            'Make sure to have the following lines on your ~/.ssh/config file:'
+            '\n\n'
+            '  Host {system_hostname}\n'
+            '      StrictHostKeyChecking no\n'
+            '      User {ssh_user}\n'
+            '      UserKnownHostsFile /dev/null\n'
+            '      ControlMaster auto\n'
+            '      ControlPersist 10m\n'
+            '      ControlPath ~/.ssh/controlmasters/%C\n'
+            .format(system_hostname=system_hostname, ssh_user=ssh_user)
+        )
+
+    click.echo('Creating the settings file at {}...'.format(path))
+    config_dict = {
+        'pulp': {
+            'auth': [pulp_username, pulp_password],
+            'version': pulp_version,
+        },
+        'systems': [{
+            'hostname': system_hostname,
+            'roles': {
+                'amqp broker': {'service': amqp_broker},
+                'api': {
+                    'scheme': system_api_scheme,
+                    'verify': system_api_verify,
+                },
+                'mongod': {},
+                'pulp celerybeat': {},
+                'pulp cli': {},
+                'pulp resource manager': {},
+                'pulp workers': {},
+                'shell': {'transport': 'ssh' if using_ssh else 'local'},
+                'squid': {},
+            }
+        }]
+    }
+    with open(path, 'w') as handler:
+        handler.write(json.dumps(config_dict, indent=2, sort_keys=True))
+    click.echo(
+        'Settings file created, run `pulp-smash settings show` to show its '
+        'contents.'
+    )
+
+
+@settings.command('show')
+@click.pass_context
+def settings_show(ctx):
+    """Show the settings file."""
+    path = ctx.obj['cfg_path']
+    if not path:
+        _raise_settings_not_found()
+
+    click.echo(
+        'Showing settings file {}\n'
+        .format(path)
+    )
```

Comment:
How about creating a separate command for showing the path to the settings file? That would let this tool be more scriptable. For example:

```console
$ pulp-smash settings show > $(pulp-smash settings path).bak
```

Here's a diff that accomplishes that:

```diff
diff --git a/pulp_smash/pulp_smash_cli.py b/pulp_smash/pulp_smash_cli.py
index 44735ce..9d866a2 100644
--- a/pulp_smash/pulp_smash_cli.py
+++ b/pulp_smash/pulp_smash_cli.py
@@ -123,6 +123,16 @@ def settings_create(ctx):
     )
 
 
+@settings.command('path')
+@click.pass_context
+def settings_path(ctx):
+    """Show the path to the settings file."""
+    path = ctx.obj['cfg_path']
+    if not path:
+        _raise_settings_not_found()
+    click.echo(path)
+
+
 @settings.command('show')
 @click.pass_context
 def settings_show(ctx):
@@ -130,11 +140,6 @@ def settings_show(ctx):
     path = ctx.obj['cfg_path']
     if not path:
         _raise_settings_not_found()
-
-    click.echo(
-        'Showing settings file {}\n'
-        .format(path)
-    )
     with open(path) as handle:
         click.echo(json.dumps(json.load(handle), indent=2, sort_keys=True))
```